SH: binutils fails to compile

[edit - related to #401 ]
Hi, just retried today with the sample sh-unknown-elf.config but enabling only big-endian instead of both; this time it's binutils complaining about "sheb-unknown-elf" :

```
[trimmed a few similar lines]
[ALL  ]    checking whether strnlen is declared... yes
[ALL  ]    checking linker --as-needed support... yes
[ALL  ]    checking for cos in -lm... yes
[ALL  ]    *** BFD does not support target sheb-unknown-elf.
[ALL  ]    *** Look in bfd/config.bfd for supported targets.
[ERROR]    make[2]: *** [Makefile:2641: configure-bfd] Error 1
[ALL  ]    make[2]: Leaving directory '~/ctng/.build/sheb-unknown-elf/build/build-binutils-host-x86_64-build_pc-linux-gnu'
[ERROR]    make[1]: *** [Makefile:849: all] Error 2
[ALL  ]    make[1]: Leaving directory '~/ctng/.build/sheb-unknown-elf/build/build-binutils-host-x86_64-build_pc-linux-gnu'
[ERROR]  
[ERROR]  >>
[ERROR]  >>  Build failed in step 'Installing binutils for host'
[ERROR]  >>        called in step '(top-level)'
```

This is with ct-ng from  git adaa3a5, and binutils-2.29.1

--- a/scripts/build/arch/sh.sh
+++ b/scripts/build/arch/sh.sh
@@ -1,8 +1,12 @@
 # Compute sh-specific values
 
 CT_DoArchTupleValues () {
-    # The architecture part of the tuple:
-    CT_TARGET_ARCH="${CT_ARCH_SH_VARIANT}${CT_ARCH_SUFFIX:-${target_endian_eb}}"
+    # The architecture part of the tuple. Binutils do not accept specifications
+    # like 'sheb-unknown-elf' even though GCC does. So keep the tuple just sh-*-elf
+    # unless user wants something specific (either CPU or explicit suffix).
+    if [ "${CT_ARCH_SH_VARIANT}" != "sh" -o -n "${CT_ARCH_SUFFIX}" ]; then
+        CT_TARGET_ARCH="${CT_ARCH_SH_VARIANT}${CT_ARCH_SUFFIX:-${target_endian_eb}}"
+    fi
 
     # Endianness stuff (uses non-standard CFLAGS). If both are compiled, let the
     # compiler's default or multilib iterator be used.

